### DISCUSSION TITLE
VarDumper shows same property twice ( with extended class )

### DISCUSSION THREAD

**Discussion by wotta** [symfony/symfony#53072] [Q&A] [answered]
Hello everyone,

I was working on some small stuff yesterday and noticed somewhat (unexpected behaviour?). Now I think a quick Q&A is the best way to share this because I am uncertain if it should be qualified as a bug.

My main question: Is this unexpected behaviour or not?
Have a nice day and thank you for reading!

### Unexpected behavior?

When you have a class that extends for example Test that has a private property called `$id`. But the current class also has a private property `$id`, in this case the vardumper shows `-id: null` twice as shown in the screenshot below.

![Screenshot 2023-12-14 at 10 26 44](https://github.com/symfony/symfony/assets/13332240/014e8805-055f-4064-8ddd-24bbdc152e72)

```php

class Test {
    private ?int $id = null;
}

class Test2 {
    private ?int $id = null;
}

dd(new Test2());
```

**Comment by MatTheCat** [comment, marked as answer]
Hello,

It’s not the same property as the base class’ is private; you can assign them different values to confirm. Maybe the VarDumper should differentiate them? I don’t know :thinking: 

You can always open a feature request and see how it goes.

**Reply by wotta** [reply]
You are completely right, both properties are "different" but I didn't except to see both with the same name to be fair.
I opened this within the `Q&A` section just to verify that is expected.

Thanks for thinking along!
